nmdc_runtime/site/ops.py

```diff
@@ -872,3 +905,41 @@ def get_csv_rows_from_url(url: Optional[str]) -> List[Dict]:
     # Collect all the rows into a list of dicts while stripping out (valfilter) cells where the
     # value is an empty string (identity returns a Falsy value).
     return [valfilter(identity, row) for row in reader]
+
+
+@op
+def get_df_from_url(url: str) -> pd.DataFrame:
```

Comment:
`@op` utilities used in the appropriate `@graph`. Basically abstracting out functionality that previously used to exist in the Translator, and using Dagster infrastructure to implement data source I/O and making sure the Translator purely has _translation_ logic.

In response to: https://github.com/microbiomedata/nmdc-runtime/pull/308#discussion_r1416102763